### PR TITLE
docs: multiple go modules rfc

### PR DESCRIPTION
**Pull request by paralta**
## Description

Add RFC to propose a new project structure which splits the main module into multiple modules to allow us to version modules separately and reduce dependencies.

## Type of Change

[ ] Bug Fix  
[ ] New Feature  
[ ] Breaking Change  
[ ] Refactor  
[x] Documentation  
[ ] Other (please describe)  

## Checklist

- [ ] I have read the [contributing guidelines](https://github.com/openclarity/vmclarity/blob/main/CONTRIBUTING.md)
- [ ] Existing issues have been referenced (where applicable)
- [ ] I have verified this change is not present in other open pull requests
- [ ] Functionality is documented
- [ ] All code style checks pass
- [ ] New code contribution is covered by automated tests
- [ ] All new and existing tests pass

### CODE REVIEW

**Review by lgecse — DISMISSED** [review]
LGTM! Thanks @paralta for putting this RFC together.

File: rfc/multiple-go-modules.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+VMClarity consists of a single module containing all the packages required to run VMClarity. However, there is a clear boundary between the following packages: api, cli, orchestrator, ui and uibackend.
+
+The build and deployment of these packages is already performed separately and independently.
+
+## Proposal
+
+The proposal here is to split the VMClarity repository into multiple modules:
+
+- **api**. Interface between all the services in VMClarity including the DB. Composed by API model, backend client and server.
+- **cli**. Responsible for running a scan in an asset and report the results back to api. Contains the logic to configure, run and manage different analysers and scanners.
```
Comment (ramizpolic) [inline rfc/multiple-go-modules.md:27]: does this mean that the cli takes ownership of scanning capabilities?

File: rfc/multiple-go-modules.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+├── docs
+├── e2e
+│   ├── config
+│   ├── go.mod
+│   └── testenv
+│       ├── docker
+│       ├── kubernetes
+│       ├── types
+│       └── utils
+├── img
```
Comment (ramizpolic) [inline rfc/multiple-go-modules.md:120]: perhaps renaming this to  `assets` would be more generalized and cleaner

File: rfc/multiple-go-modules.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+├── cli
+│   ├── cmd
+│   ├── go.mod
+│   ├── pkg
+│   │   ├── analyzer
+│   │   ├── config
+│   │   ├── converter
+│   │   ├── families
+│   │   ├── findingkey
+│   │   ├── job_manager
```
Comment (ramizpolic) [inline rfc/multiple-go-modules.md:104]: should be `jobmanager` since the table above lists it properly

**Review by ramizpolic — APPROVED** [review]
lgtm! :rocket:

File: rfc/multiple-go-modules.md (hunk 1 above)
Comment (paralta) [reply rfc/multiple-go-modules.md:27]: Yes, it does. And at some point we might want to remove the scanner from the CLI, we can address this on the scope of https://github.com/openclarity/vmclarity/issues/346.

File: rfc/multiple-go-modules.md (hunk 3 above)
Comment (paralta) [reply rfc/multiple-go-modules.md:104]: thanks for noticing this!

**Comment by paralta** [comment]
@ramizpolic @lgecse thank you so much for reviewing 😄

**Review by lgecse — DISMISSED** [review]
LGTM! :)

**Review by ramizpolic — APPROVED** [review]

File: rfc/multiple-go-modules.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    ├── log
+    ├── manifest
+    └── version
+```
+
+# Release
+
+Each module will have a tag with the format `prefix/version` where prefix is the directory within the repository where the module is defined, more details [here](https://go.dev/wiki/Modules#publishing-a-release). For now, the same version will be used for each module even if there are no changes, this will simplify managing compatibility between modules across versions.
+
+The release GitHub workflow will be extended with an additional step to create a tag for each module. This step will only be performed after the release is successfully published.
```
Comment (chrisgacsal) [inline rfc/multiple-go-modules.md:184]: I think we need to include a plan for the release process as there are a couple of open questions still:

Releasing all the modules in one go would work using `replace` directives for local modules packages in `go.mod` for modules in the repository. However the release process also needs to ensure that the module versions are bumped in `go.mod` as `replace` directives are not considered when a modules is being imported by 3rd party.

File: rfc/multiple-go-modules.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+| uibackend/server | pkg/uibackend/rest            | uibackend/server                 | uibackend/server/v0.7.0                 |
+| uibackend/client | pkg/shared/uibackendclient    | uibackend/client                 | uibackend/client/v0.7.0                 |
+| utils            | pkg/version                   | utils/version                    | utils/v0.7.0                            |
+| utils            | pkg/shared/command            | utils/command                    | utils/v0.7.0                            |
+| utils            | pkg/shared/fsutils            | utils/fsutils                    | utils/v0.7.0                            |
+| utils            | pkg/shared/log                | utils/log                        | utils/v0.7.0                            |
+| utils            | pkg/shared/manifest           | utils/manifest                   | utils/v0.7.0                            |
+| provider         | pkg/orchestrator/provider     | provider                         | provider/v0.7.0                         |
+| provider         | pkg/containerruntimediscovery | provider/common/runtimediscovery | provider/common/runtimediscovery/v0.7.0 |
```
Comment (chrisgacsal) [inline rfc/multiple-go-modules.md:63]: I would move the `e2e/testenv` into a separate module so we can reuse it outside the `e2e` too.

Comment (paralta) [reply rfc/multiple-go-modules.md:63]: @chrisgacsal what do you think about adding a test folder with the two modules (e2e and testenv) inside? Like
```
| Module       | Current path  | New path     | Version Tag         |
| ------------ | ------------- | ------------ | ------------------- |
| test/e2e     | e2e/          | test/e2e     | test/e2e/v0.7.0     |
| test/testenv | e2e/testenv   | test/testenv | test/testenv/v0.7.0 |
```

File: rfc/multiple-go-modules.md (hunk 4 above)
Comment (paralta) [reply rfc/multiple-go-modules.md]: just pushed a commit extending the release process changes, let me know what you think 😃

File: rfc/multiple-go-modules.md (hunk 5 above)
Comment (paralta) [reply rfc/multiple-go-modules.md:63]: added 👍

**Review by chrisgacsal — APPROVED** [review]
LGTM! Great job @paralta!

**Pull request merged**